sqlmesh/core/plan/builder.py

```diff
@@ -577,8 +577,14 @@ def _apply_effective_from(self) -> None:
             if not snapshot.disable_restatement:
                 snapshot.effective_from = self._effective_from
 
-    def _is_forward_only_model(self, s_id: SnapshotId) -> bool:
+    def _is_forward_only_change(self, s_id: SnapshotId) -> bool:
         snapshot = self._context_diff.snapshots[s_id]
+        if snapshot.name in self._context_diff.modified_snapshots:
+            _, old = self._context_diff.modified_snapshots[snapshot.name]
+            # If the model kind has been changed from non-materialized to materialized,
+            # then we should not consider this to be a forward-only change.
+            if not old.is_materialized:
```

Comment:
can we change any kind in a forward only way? incremental -> unique by key -> scd type 2 ???

Reply:
That is a good point. Let me think about this more and make a change accordingly.

Reply:
Updated